### PR TITLE
Version 1.0.7 changes documentation & GitHub action workflow for automated releases

### PR DESCRIPTION
**Pull request by alephnull7**
The only change to the normal repo items is to `CHANGELOG.md` and summarize the changes since version `1.0.6`.

Added is a custom workflow that automates releases to the repo when changes are performed on `CHANGELOG.md`. In summary, the most recent version change listed (the first `####` header encountered) in combination with the current UTC timestamp are used to update version and date information in `DESCRIPTION`, core R files (those in `R/`) that have a `#version` comment, and also in `CHANGELOG.md` to make sure the version and timestamp are the same everywhere. After this, PACVr is built with `R CMD build`, and added as a release asset, in addition to the standard repo source code assets. The default "body" text of the release is that of the details corresponding to this version (the unordered list specified by `*`s following the first `####` header).

TL;DR
If you add a new version and related description to `CHANGELOG.md`, all timestamps will be updated to "now", and all mentions of the package version will reflect this stated version. After this happens, a release will be created corresponding to this version, which will include an R source package reflecting the current state of the repo.

### CODE REVIEW

**Comment by michaelgruenstaeudl** [comment]
That is a great workflow that will come in very handy over the next weeks and months!
